util/events/libuv/Pipe.c

```diff
@@ -392,6 +393,61 @@ static struct Pipe_pvt* newPipe(struct EventBase* eb,
     return out;
 }
 
+static struct Pipe_pvt* newPipeAny(struct EventBase* eb,
+                                  const char* fullPath,
+                                  struct Except* eh,
+                                  struct Allocator* userAlloc)
+{
+    struct EventBase_pvt* ctx = EventBase_privatize(eb);
+    struct Allocator* alloc = Allocator_child(userAlloc);
+
+    char* name = Allocator_malloc(alloc, (fullPath ? CString_strlen(fullPath) : 0));
+    if (fullPath) {
+        Bits_memcpy(name, fullPath, CString_strlen(fullPath));
+        name = basename(name);
+    }
```

Comment:
If fullPath is null, you want name to be null ?   I'm thinking Allocator_malloc() might not work that way...
What about:
```
char* name = NULL;
if (fullPath) {
    name = basename(String_new(fullPath, alloc)->bytes);
}
```

Reply:
Does it make sense to throw an exception if fullPath is null?